### PR TITLE
feat: dump application router json

### PR DESCRIPTION
**Pull request by fengmk2**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

Demo of router.json https://gist.github.com/fengmk2/7f7df8c33cf36b8685a3196ca50656a8

### CODE REVIEW

**Comment by fengmk2** [comment]
2个 commits，第一个是bugfix

File: lib/application.js [hunk 1]
```diff
@@ -115,6 +116,34 @@ class Application extends EggApplication {
...
+      const dumpRouterFile = path.join(rundir, 'router.json');
+      const routers = [];
+      for (const layer of this.router.stack) {
+        routers.push({
+          name: layer.name,
+          methods: layer.methods,
+          paramNames: layer.paramNames,
+          path: layer.path,
+          regexp: layer.regexp.toString(),
+          stackLength: layer.stack.length,
```
Comment (fengmk2) [inline lib/application.js]: 这个我完善一下

**Comment by fengmk2** [comment]
还依赖 egg-core 的 pr

**Comment by fengmk2** [comment]
deps on https://github.com/eggjs/egg-core/pull/131

File: lib/application.js [hunk 2]
```diff
@@ -115,6 +116,35 @@ class Application extends EggApplication {
     return context;
   }
 
+  /**
+   * save app.config to `run/${type}_config.json`
```
Comment (atian25) [inline lib/application.js]: - 几个地方的 dumpConfig 感觉都可以改名为 `dumpRuntime` 或 `dumpMeta` 了。
- jsdoc 也要改下。

**Comment by atian25** [comment]
这个可以合并了?

**Comment by fengmk2** [comment]
@atian25 ci 过了就合并发布。

**Comment by fengmk2** [comment]
https://github.com/eggjs/egg/pull/1758

**Comment by fengmk2** [comment]
1.12.0

**Pull request merged**